engine/runtime/src/main/java/org/enso/interpreter/node/callable/InvokeMethodNode.java

```diff
@@ -349,6 +356,35 @@ Stateful doConvertDateTime(
     }
   }
 
+  @Specialization(
+      guards = {
+        "!types.hasType(self)",
+        "!types.hasSpecialDispatch(self)",
+        "getPolyglotCallType(self, symbol.getName(), interop) == CONVERT_TO_DURATION"
+      })
+  Stateful doConvertDuration(
+      VirtualFrame frame,
+      Object state,
+      UnresolvedSymbol symbol,
+      Object self,
+      Object[] arguments,
+      @CachedLibrary(limit = "10") TypesLibrary types,
+      @CachedLibrary(limit = "10") InteropLibrary interop,
+      @Cached MethodResolverNode methodResolverNode) {
+    var ctx = Context.get(this);
+    try {
+      var duration = interop.asDuration(self);
+      var ensoDuration = new EnsoDuration(duration);
```

Comment:
This is exact copy of date, time & co. handling. However in case of _array_ handling I decided to not do any wrapping, just let the `hasArrayElements` object flow thru the engine. Pavel, what do you think is better from Truffle perspective: wrap or let the original `TruffleObject` flow?

If _"let the original object flow"_, can you try it with duration and other `doConvertXYZ` methods as well?